From c2f434f1cf05b02caac37d44564641943ebbcb26 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:03:41 -0400
Subject: [usb] wusbcore: fix panic in wusbhc_chid_set

Message-id: <1410887463-20674-150-git-send-email-dzickus@redhat.com>
Patchwork-id: 95498
O-Subject: [RHEL7 PATCH 149/591] usb: wusbcore: fix panic in wusbhc_chid_set
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit bd130adacaf8cea179f9a700fb694f5be3b05bf0
Author: Thomas Pugliese <thomas.pugliese@gmail.com>
Date:   Wed Apr 23 14:28:10 2014 -0500

    usb: wusbcore: fix panic in wusbhc_chid_set

    If no valid CHID value has previously been set on an HWA, writing a
    value of all zeros will cause a kernel panic in uwb_radio_stop because
    wusbhc->uwb_rc has not been set.  This patch skips the call to
    uwb_radio_stop if wusbhc->uwb_rc has not been initialized.

    Signed-off-by: Thomas Pugliese <thomas.pugliese@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/wusbcore/mmc.c b/drivers/usb/wusbcore/mmc.c
index 4474126..3f485df 100644
--- a/drivers/usb/wusbcore/mmc.c
+++ b/drivers/usb/wusbcore/mmc.c
@@ -301,7 +301,7 @@ int wusbhc_chid_set(struct wusbhc *wusbhc, const struct wusb_ckhdid *chid)
 
  if (chid)
   result = uwb_radio_start(&wusbhc->pal);
- else
+ else if (wusbhc->uwb_rc)
   uwb_radio_stop(&wusbhc->pal);
 
  return result;
-- 
1.7.1